fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneIndexValidator.java

```diff
@@ -44,5 +46,54 @@ public void validate(@Nonnull MetaDataValidator metaDataValidator) {
         for (RecordType recordType : metaDataValidator.getRecordMetaData().recordTypesForIndex(index)) {
             LuceneIndexExpressions.validate(index.getRootExpression(), recordType.getDescriptor());
         }
+
+        validateIndexOptions(index);
+    }
+
+    @VisibleForTesting
+    public static void validateIndexOptions(@Nonnull Index index) {
+        validateAnalyzerNamePerFieldOption(LuceneIndexOptions.LUCENE_ANALYZER_NAME_PER_FIELD_OPTION, index);
+        validateAnalyzerNamePerFieldOption(LuceneIndexOptions.AUTO_COMPLETE_ANALYZER_NAME_PER_FIELD_OPTION, index);
+        validateAutoCompleteExcludedFields(index);
+    }
+
+    private static void validateAnalyzerNamePerFieldOption(@Nonnull String optionKey, @Nonnull Index index) {
+        String analyzerNamePerFieldOption = index.getOption(optionKey);
+        if (analyzerNamePerFieldOption != null) {
+            analyzerNamePerFieldOption = analyzerNamePerFieldOption.strip();
+            if (analyzerNamePerFieldOption.startsWith(LuceneIndexOptions.DELIMITER_BETWEEN_ELEMENTS)
+                    || analyzerNamePerFieldOption.endsWith(LuceneIndexOptions.DELIMITER_BETWEEN_ELEMENTS)) {
+                throw new MetaDataException("Index " + index.getName() + " has invalid option value for " + optionKey + ": " + analyzerNamePerFieldOption);
+            }
+            final String[] keyValuePairs = analyzerNamePerFieldOption.split(LuceneIndexOptions.DELIMITER_BETWEEN_ELEMENTS);
+            for (int i = 0 ; i < keyValuePairs.length; i++) {
+                final String keyValue = keyValuePairs[i].strip();
+                int firstIndex = keyValue.indexOf(LuceneIndexOptions.DELIMITER_BETWEEN_KEY_AND_VALUE);
+                if (keyValue.isEmpty()
+                        || firstIndex < 1 || firstIndex > keyValue.length() - 2
+                        || firstIndex != keyValue.lastIndexOf(LuceneIndexOptions.DELIMITER_BETWEEN_KEY_AND_VALUE)) {
+                    throw new MetaDataException("Index " + index.getName() + " has invalid option value for " + optionKey + ": " + analyzerNamePerFieldOption);
```

Comment:
Should this validate that the (stripped) key and value are non-empty?

Reply:
I think empty key or value should cause exception because the check ```keyValue.isEmpty() || firstIndex < 1 || firstIndex > keyValue.length() - 2```

Reply:
Oh, I see. It's a little complicated, but I think you're right: when the `keyValue` is stripped, if there's a completely blank key or value, then either `firstIndex` will be < 1 (for a blank key) or `firstIndex` will be > length - 2 (for a blank value), assuming there's only one delimiter. But I had to convince myself that was enough